x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -110,7 +117,27 @@ export class Embeddable
 
     this.expressionRenderer = deps.expressionRenderer;
     this.initializeSavedVis(initialInput).then(() => this.onContainerStateChanged(initialInput));
-    this.subscription = this.getInput$().subscribe((input) => this.onContainerStateChanged(input));
+
+    const input$ = this.getInput$();
+    this.subscription = input$.subscribe((input) => this.onContainerStateChanged(input));
+    input$
```

Comment:
Can we leave a little comment here explaining the reasoning? I can imagine we have to touch this at some point down the line

Reply:
Added a comment.